### PR TITLE
vm: Flag names (refactoring) 

### PR DESCRIPTION
**Pull request by VelocityRa**
All the magic/unnamed values in vm code made it even more difficult to try and understand.
Hopefully this improves things.

Some names are missing, suggestions welcome.

### CODE REVIEW

**Comment by Nekotekina** [comment]
sudo_mirror -> preallocated or nonshareable

File: rpcs3/Emu/Memory/vm.cpp [hunk 1]
```diff
@@ -1515,17 +1519,17 @@ namespace vm
...
 				}
 
-				if (!must_be_empty && ((*it)->flags & 0x3) != 2)
+				if (!must_be_empty && ((*it)->flags & bf0_mask) != bf0_0x2)
 				{
 					continue;
 				}
 
-				if (must_be_empty && (it->use_count() != 1 || (*it)->imp_used(lock)))
+				if (must_be_empty && (it->use_count() != bf0_0x1 || (*it)->imp_used(lock)))
```
Comment (Nekotekina) [inline rpcs3/Emu/Memory/vm.cpp:1532]: Not a flag, use_count is compared to self-describing constant.

File: rpcs3/Emu/Memory/vm.h [hunk 2]
```diff
@@ -90,6 +90,31 @@ namespace vm
 	// utils::memory_lock wrapper for locking sudo memory
 	void lock_sudo(u32 addr, u32 size);
 
+	enum block_flags_3
+	{
+		block_rw = 0x1000,
```
Comment (Nekotekina) [inline rpcs3/Emu/Memory/vm.h:95]: It means pages are inaccessible

Comment (VelocityRa) [reply rpcs3/Emu/Memory/vm.h:95]: shit yea

Comment (Nekotekina) [reply rpcs3/Emu/Memory/vm.h:95]: Or better to say, hidden pages

**Comment by Nekotekina** [comment]
Last ones I forgot myself, sadly.

**Pull request merged**